cytoxnet/dataprep/dataprep.py

```diff
@@ -1,2 +1,156 @@
 # potentially useful: deepchem dataset class or pandas split
 # having our data in the deepchem.dataset form at the end is very helpful
+
+
+"""
+Data splitting module for creating train/test splits  
+"""
+
+import pandas as pd
+import deepchem
+import sklearn
+
+
+def convert_cats(dataframe):
+    """
+    Drops rows with missing values and converts categorical values to integers.
+    
+    Parameters
+    ----------
+    dataframe: featurized dataframe
+    
+    Returns
+    -------
+    modified dataframe without any na values in the 'Mol' column 
+    and all categorical columns with string values converted to integer values
+    """
+    # Drop all na values in the 'Mol' column
+    # May want to expand the range of columns beyond 'Mol'
+    
+    dataframe = dataframe.dropna()
+
+    # convert categorical values, if any exist, into integers
+    # might be able to use sklearn Label Encoder or OnHotEncoder
+    # might be necessary if we categorize target areas/tox methods/etc.
+    # Iterate through columns and convert any containing strings to ints
+    for column in dataframe:
+    	column_series_obj = dataframe[column]
+    	if dataframe[column].dtype not numeric:
+    		l = sklearn.preprocessing.LabelEncoder()
+    		l.fit(dataframe[column])
+    		dataframe.column = l.transform(dataframe.column)
+    	else continue
+    return dataframe
+
+
+
+
+
+
+def trans_data(dataframe, X, y, w, ids, 
+	transform_list: list = ['Normalize',], to_transform: str = 'X', **kwargs):
+    """
+    Converts featurized dataframe into a transformed dataset object.
+    
+    Parameters
+    ----------
+   	dataframe: featurized dataframe 
+   	X: (str or List[str]) name(s) of the column(s) containing the X array.
+   	y: (str or List[str]) name(s) of the column(s) containing the y array.
+   	w: (str or List[str]) name(s) of the column(s) containing the w array.
+   	ids: (str) name of the column containing the ids.
+   	transform_list: (List[str]) list of transformation methods to pass dataset through
+   	to_transform: (str) specification of whether to transform X, y, or w
+   	**kwargs: key word arguments to be passed to selected transformer
+
+
+    Returns
+    -------
+    transformed dataset object
+    """
+    # convert dataframe to dataset using deepchem.data.NumpyDataset function
+    to_dataset = deepchem.data.NumpyDataset(X, y, w, ids)
+	dataset = to_dataset.from_dataframe(dataframe)
+
+    if to_transform == 'X':
+    	transform_X = True
+        transform_y = False
+   	elif to_transform == 'y':
+        transform_X = False
+        transform_y = True
+
+    # feed dataset into list of transformers sequentially, returning a single transformed dataset
+   	for transform in transform_list:
```

Comment:
I don't think this should be part of the if statement

Reply:
Do you think it should be it's own if statement rather than an elif?  Could you clarify why lines 79 and 80 should not be part of the if statement?

Reply:
I am referring the to line the comment is on - 83 - its currently part of the if based on indentation

Reply:
I think I fixed that in the most recent push